File: VB/E5040/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports E5040.Reports

Namespace E5040.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"

			Return View()
		End Function

		Public Function CallbackPanelPartial(ByVal fieldName As String, ByVal isShiftKeyPressed As Boolean) As ActionResult
			ProductsReport.SortBy(fieldName, isShiftKeyPressed)
			Return PartialView("_CallbackPanelPartial")
		End Function

		Public Function DocumentViewerPartial() As ActionResult
			Dim report As New ProductsReport()
			Return PartialView("_DocumentViewerPartial", report)
		End Function

		Public Function DocumentViewerExport() As ActionResult
			Dim report As New ProductsReport()
			Return ReportViewerExtension.ExportTo(report)
		End Function
	End Class
End Namespace

File: CS/E5040/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using E5040.Reports;

namespace E5040.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";

            return View();
        }

        public ActionResult CallbackPanelPartial(string fieldName, bool isShiftKeyPressed) {
            ProductsReport.SortBy(fieldName, isShiftKeyPressed);
            return PartialView("_CallbackPanelPartial");
        }

        public ActionResult DocumentViewerPartial() {
            ProductsReport report = new ProductsReport();
            return PartialView("_DocumentViewerPartial", report);
        }

        public ActionResult DocumentViewerExport() {
            ProductsReport report = new ProductsReport();
            return ReportViewerExtension.ExportTo(report);
        }
    }
}